### ISSUE TITLE
Mod feature: IP ban type “No registration”

### ISSUE THREAD

**Issue by kittey** [danbooru/danbooru#5122]
Please add a new IP ban type that disallows registrations but allow all other action from that IP (range).

Currently we have the types “Partial” and “Full”. “Partial” is but a minor annoyance for persistent trolls because free email addresses are readily available. “Full” is good if a user has a static IP or always uses the same range/set of addresses. Especially mobile users are assigned addresses from a large range/pool of addresses and banning that potentially causes a lot of collateral damage by locking out legitimate users.

Please add a third ban type that’s in-between: disallowing only registrations but allowing everything else. This will be more effective at locking out trolls than a partial ban while not causing any trouble for existing users.

The “Partial” type should be renamed to something like “Mail required” to be more descriptive.

**Comment by nonamethanks** [comment]
I second this. It would be extremely useful since now we have to basically blanket ban entire proxies because of a few uncontrolled autists.

**Comment by kittey** [comment]
> because of a few uncontrolled autists.

That’s an insult to autists… >_>

**Comment by nonamethanks** [comment]
Well, let's call them turbo autists then. Especially [this guy](https://danbooru.donmai.us/ip_bans?commit=Search&search%5Breason_like%5D=%2A838083%2A), that cycles through proxies until they're banned. I had to unban a few because they actually hit other people that had a legitimate use for them, so if we could just stop registrations from there it would help immensely.

Maybe we could also consider barring registration from proxies/TOR by default? Though not sure if that's desiderable.

**Comment by kittey** [comment]
Disallowing registration from proxies/TOR would be a disservice to users from non-free states and those from states that are censoring Danbooru.

When I investigated some (member-tier) account sharing, I actually received a few messages from users saying that they can’t access Danbooru normally because of state censorship.

**Comment by nonamethanks** [comment]
Yeah, that's fair. I think we're banned in most of Russia for example, wouldn't make much sense to block it all.

**Comment by BrokenEagle** [comment]
I'm just trying to think of ways that can still facilitate legitimate users while also blocking the bad ones, so how about allowing users underneath a blanket registration IP ban the ability to register with a one-time passcode (OTP) provided by a Moderator or Admin? Then those users would have to convince a person that they aren't one of those trolls in order to be able to register, thus upping the difficulty level. The OTP also would be registered against whoever created it, and so they would be on the hook, as it were, if the person they gave it to turns out to be malicious.

**Comment by kittey** [comment]
> I'm just trying to think of ways that can still facilitate legitimate users while also blocking the bad ones […]

I’d like Gold+ user to be able to send out invitation mails containing a link (with OTP, I guess) that will allow the recipient to register despite using a registration-banned IP. Invitation mails would be limited to maybe 1/week and invitation permission should be removable. My guess is that the few untrustworthy users would be weeded out quickly and then there’s no need to bother mods/admins with invitations anymore.

**Comment by BrokenEagle** [comment]
Gold+ would probably be acceptable, to me at least, as that level has something to lose if they keep allowing the registration of bad actors. Member-level not so much though, and the account age thing doesn't mean much, since there are probably accounts on Danbooru pre-2010 which have probably never been used, for example. But I imagine that most Member users would know at least one Gold+ user, and if not, they can always hop on Discord and plead their case.

**Comment by kittey** [comment]
Revisiting my suggestion, I actually favor BrokenEagle’s passcode suggestion over sending out emails, making it a bit more involved.

Maybe passcodes should also not be available via the API, though I have to admit that that won’t change much.
